fastly/custom_tls_certificate.go

```diff
@@ -0,0 +1,195 @@
+package fastly
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/google/jsonapi"
+)
+
+// CustomTLSCertificate represents a custom certificate. Uses common TLSDomain type from BulkCertificate.
+type CustomTLSCertificate struct {
+	ID                 string       `jsonapi:"primary,tls_certificate"`
+	IssuedTo           string       `jsonapi:"attr,issued_to"`
+	Issuer             string       `jsonapi:"attr,issuer"`
+	Name               string       `jsonapi:"attr,name"`
+	NotAfter           *time.Time   `jsonapi:"attr,not_after,iso8601"`
+	NotBefore          *time.Time   `jsonapi:"attr,not_before,iso8601"`
+	Replace            bool         `jsonapi:"attr,replace"`
+	SerialNumber       string       `jsonapi:"attr,serial_number"`
+	SignatureAlgorithm string       `jsonapi:"attr,signature_algorithm"`
+	TLSDomains         []*TLSDomain `jsonapi:"relation,tls_domains"`
+	CreatedAt          *time.Time   `jsonapi:"attr,created_at,iso8601"`
```

Comment:
Was going to ask for the jsonapi 'attributes' to be grouped and ordered, but I then recalled @phamann telling me this is a pre-existing convention used, where 'metadata' related fields are grouped at the bottom of the struct. 

Personally I'm not sure if I'm a fan of it, as I find it disorientating when comparing the code to the actual API documentation, but for the sake of forward progress I'm not going to 'bikeshed' this change 🙂

Reply:
Yeah. I originally put them in the order of the JSON API (which I believe is alphabetical) but @phamann asked me to move the CreatedAt and UpdatedAt fields to the bottom.